src/main/kotlin/org/jitsi/jicofo/xmpp/ConferenceIqHandler.kt

```diff
@@ -104,9 +104,12 @@ class ConferenceIqHandler(
     private fun processExtensions(query: ConferenceIq, response: ConferenceIq?, roomExists: Boolean): IQ? {
         val peerJid = query.from
         var identity: String? = null
+        val room = query.room
+        val breakoutRoomRegex = Regex("#breakout_[-\\da-f]{36}@${XmppConfig.client.conferenceMucJid}$")
+        val isBreakoutRoom = room.toString().contains(breakoutRoomRegex)
```

Comment:
Does this allow users to bypass authentication by using a special room name?

Reply:
Thank you very much @bgrozev for your thorough review. You are right, special breakout room names will bypass the authentication check. This is to allow users to join a breakout room without having a moderator to join the room first. Authorisation is checked [here](https://github.com/jitsi/jitsi-meet/pull/9131/files#diff-844d74e9c13f71420b4cba715cd6167cfae74192cf534166c4e04b8b41001512R230) on Prosody by rejecting any breakout room which is not valid. So more concrete: Prosody will reject the creation of the room on the muc-room-pre-create event. Security is achieved by the random uuid in the breakout room names.

Reply:
Sorry for the slow response, I missed that! Prosody only rejects the creation of a breakout room when `mod_breakout_rooms` is loaded, correct? Without it anyone can create a room that matches `breakoutRoomRegexp` and bypass jicofo's authentication?

Reply:
Good catch, @bgrozev! 

I see two options to get this fixed:

1. make the main muc in prosody block rooms with the breakout room name scheme by default and have the breakout rooms module allow these with a higher priority.
1. re-introduce  a new breakout room muc component

I tend to prefer the first option as it avoids the overhead of introducing a breakout domain handling to jitsi-meet, lib-jitsi-meet and jicofo. But I admit that the second solution would allow a clearer distinction between the room types. So on the long run we might need to go down that road. I would just like to try to avoid setting the hurdle to hight for the initial release of this feature.

Or do you have another idea? What do you think?

Reply:
Can you elaborate on option 2? How would a new room muc component helps us fix the problem?

Reply:
It would mostly come down to roll back these changes:

- https://github.com/jitsi/jitsi-meet/pull/9131/commits/02888917e463eab054f9b9d3257189eea284fd18
- https://github.com/jitsi/jitsi-meet/pull/9131/commits/e40f76287592af013854e14426e6a26a4f312035
- https://github.com/jitsi/lib-jitsi-meet/pull/1654/commits/26ef339f6c4a3e742f55029e375dc83e1ee467ee
- https://github.com/jitsi/docker-jitsi-meet/pull/1088/commits/faf72082ae986d51d719d12f84cbc5f0e3e0a1b7
- jicofo: https://github.com/jitsi/jicofo/pull/759/commits/4a990db4b4862c1c8432736f59f0a6dd20a3318b

So actually it was like that before. However comments about the introduction of a new muc and the handling of the domain name made me consider to renounce to that approach.

This would fix the problem as only the breakout module would handle breakout mucs. So there would be no chance for somebody to start a breakout room without the breakout module being active (and checking whether it is a valid room).

Reply:
> This would fix the problem as only the breakout module would handle breakout mucs. So there would be no chance for somebody to start a breakout room without the breakout module being active (and checking whether it is a valid room).

That sounds quite good actually. Can you elaborate on the drawbacks? Sorry for the back and forth here :-/

Reply:
Well, drawbacks come down to what is in those patches. That is:

- We will need to teach jitsi-meet, lib-jitsi-meet, prosody and jicofo the new muc domain (in contrast to only let them know about the "weird" breakout room names)
- The bit more complicated configuration of Prosody
- @saghul asking to make the domain name configurable :D

Reply:
That saghul guy.... :-P

Let's try to brainstorm about this next week and figure out a way forward.